patches/server/1044-add-PlayerBrushBlockEvent.patch

```diff
@@ -0,0 +1,51 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: chmodsayshello <chmodsayshello@hotmail.com>
+Date: Wed, 1 Nov 2023 14:01:08 +0100
+Subject: [PATCH] add PlayerBrushBlockEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/item/BrushItem.java b/src/main/java/net/minecraft/world/item/BrushItem.java
+index d2ae20ec96a51c5d217c3695d53eda492f9090bc..d91c839491af81d53c0cc6beab2880419ae8ca99 100644
+--- a/src/main/java/net/minecraft/world/item/BrushItem.java
++++ b/src/main/java/net/minecraft/world/item/BrushItem.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.world.item;
+ 
++import io.papermc.paper.event.player.PlayerBrushBlockEvent;
+ import net.minecraft.core.BlockPos;
+ import net.minecraft.core.Direction;
+ import net.minecraft.core.particles.BlockParticleOption;
+@@ -26,6 +27,7 @@ import net.minecraft.world.level.block.state.BlockState;
+ import net.minecraft.world.phys.BlockHitResult;
+ import net.minecraft.world.phys.HitResult;
+ import net.minecraft.world.phys.Vec3;
++import org.bukkit.Location;
+ 
+ public class BrushItem extends Item {
+     public static final int ANIMATION_DURATION = 10;
+@@ -39,8 +41,14 @@ public class BrushItem extends Item {
+     @Override
+     public InteractionResult useOn(UseOnContext context) {
+         Player player = context.getPlayer();
++        // Paper start - PlayerBrushBlockEvent
+         if (player != null && this.calculateHitResult(player).getType() == HitResult.Type.BLOCK) {
+-            player.startUsingItem(context.getHand());
++            io.papermc.paper.event.player.PlayerBrushBlockEvent event = new PlayerBrushBlockEvent((org.bukkit.entity.Player) player.getBukkitEntity(), org.bukkit.craftbukkit.block.CraftBlock.at(context.getLevel(), context.getClickedPos()));
++            event.callEvent();
++            if (!event.isCancelled()) {
++                player.startUsingItem(context.getHand());
++            }
++            // Paper end
+         }
+ 
+         return InteractionResult.CONSUME;
+@@ -86,8 +94,7 @@ public class BrushItem extends Item {
+                             BlockEntity var18 = world.getBlockEntity(blockPos);
+                             if (var18 instanceof BrushableBlockEntity) {
+                                 BrushableBlockEntity brushableBlockEntity = (BrushableBlockEntity)var18;
+-                                boolean bl2 = brushableBlockEntity.brush(world.getGameTime(), player, blockHitResult.getDirection());
+-                                if (bl2) {
++                                if (brushableBlockEntity.brush(world.getGameTime(), player, blockHitResult.getDirection())) { // Paper - remove "bl2" boolean (already defined)
```

Comment:
To clarify: even though I did not change this function, Paper didn't compile due to being another variable within the same scope in there. Thats why I had to make this change...

Reply:
Decompile fixes can go into the mc dev fixes patch

Reply:
done, thanks!